pkg/logql/syntax/ast_test.go

```diff
@@ -681,3 +681,29 @@ func TestParseLargeQuery(t *testing.T) {
 	_, err := ParseExpr(line)
 	require.NoError(t, err)
 }
+
+func TestGroupingString(t *testing.T) {
+	g := Grouping{
+		Groups:  []string{"a", "b"},
+		Without: false,
+	}
+	require.Equal(t, " by (a,b)", g.String())
+
+	g = Grouping{
+		Groups:  []string{},
+		Without: false,
+	}
+	require.Equal(t, "", g.String())
```

Comment:
I would assume this structure is equivalent to `by ()` 🕵️‍♀️ 
the empty grouping should be the one where `Grouping = nil`, no?

Reply:
The code was changed to handle `Grouping = nil` and the tests updated.